erpnext/hr/doctype/salary_slip/salary_slip.py

```diff
@@ -193,21 +197,37 @@ def get_leave_details(self, joining_date=None, relieving_date=None, lwp=None, fo
 			return
 
 		holidays = self.get_holidays_for_employee(self.start_date, self.end_date)
-		actual_lwp = self.calculate_lwp(holidays, working_days)
+		non_working_days = 0
+		if payroll_based_on == 'Leave':
+			actual_lwp = self.calculate_lwp(holidays, working_days)
+		elif payroll_based_on == 'Attendance':
+			actual_absent_days = self.calculate_absents()
+		else:
+			frappe.throw(_("Please set <b>Payroll based on</b> in <b>HR settings</b>"))
```

Comment:
Don't use HTML tags inside message, it creates problem in translation

Reply:
Done